forc/src/ops/forc_explorer.rs

```diff
@@ -0,0 +1,149 @@
+use std::fs::{create_dir_all, remove_file, rename, File};
+use std::io::Cursor;
+use std::path::PathBuf;
+
+use dirs;
+use reqwest;
+use serde::Deserialize;
+use tar::Archive;
+use warp::Filter;
+
+use crate::cli::ExplorerCommand;
+
+type DownloadResult<T> = std::result::Result<T, Box<dyn std::error::Error + Send + Sync>>;
+
+#[derive(Deserialize, Debug)]
+struct GitHubRelease {
+    url: String,
+    assets: Vec<GitHubReleaseAsset>,
+}
+
+#[derive(Deserialize, Debug)]
+struct GitHubReleaseAsset {
+    browser_download_url: String,
+}
+
+const REPO_RELEASES_URL: &str = "https://api.github.com/repos/FuelLabs/block-explorer-v2/releases";
+
+struct EndPoints {}
+
+impl EndPoints {
+    pub fn static_files() -> String {
+        "static".to_string()
+    }
+}
+
+struct ExplorerAppPaths {}
+
+impl ExplorerAppPaths {
+    pub fn web_app_path() -> PathBuf {
+        dirs::home_dir().unwrap().join(".fuel/explorer")
+    }
+    pub fn build_archive_path() -> PathBuf {
+        dirs::home_dir().unwrap().join(".fuel/explorer/build.tar")
+    }
+    pub fn build_archive_unpack_path() -> PathBuf {
+        dirs::home_dir().unwrap().join(".fuel/explorer/build")
+    }
+    pub fn web_app_files_path() -> PathBuf {
+        dirs::home_dir().unwrap().join(".fuel/explorer/www")
+    }
+    pub fn web_app_static_assets_path() -> PathBuf {
+        dirs::home_dir().unwrap().join(".fuel/explorer/www/static")
+    }
+}
+
+pub(crate) async fn exec(command: ExplorerCommand) -> Result<(), reqwest::Error> {
+    let ExplorerCommand { port } = command;
+    if !has_static_files() {
```

Comment:
I'm curious, and I'm sorry if I'm missing something obvious: what happens if we do have all the static files already installed locally but there's a newer version of the explorer and I want to update it?

Reply:
yeah, you'd need to delete the previous static files. One improvement that I can think of is versioning and storing the static files in directories like `~/.fuel/explorer/<X.X.X>/www`. This way `fuel explorer` would serve the highest version available and it would allow a potential new feature, running a custom version: `forc explorer --version X.X.X`

Reply:
Seems reasonable to me

Reply:
That sounds like a great approach!